[Outlook] (preview) Add-in support in the new Outlook on Windows (#3837)
OfficeDev/office-js-docs-pr: docs/outlook/one-outlook.md

---
title: Develop Outlook add-ins for the new Outlook on Windows (preview)
description: Learn about the new Outlook on Windows that's in preview and how to develop Outlook add-ins that are compatible with it.
ms.topic: article
ms.date: 02/07/2023
ms.localizationpriority: medium
---

# Develop Outlook add-ins for the new Outlook on Windows (preview)

The new Outlook on Windows desktop client unifies the Windows and web codebases to create a more consistent Outlook experience for users and administrators. Its modern and simplified interface has added capabilities and aims to improve productivity, organization, and collaboration for users. More importantly, the new Outlook on Windows supports Outlook web add-ins, so that you can continue to extend Outlook's functionality.

## Impact on VSTO and COM add-ins

> Because the new Outlook on Windows is based on the Outlook on the web codebase, VSTO and COM add-ins aren't supported. To ensure your add-in continues to work in the new Outlook on Windows, you must migrate your VSTO or COM add-in to an Outlook web add-in. Migrating to an Outlook web add-in not only enables compatibility with the new Outlook on Windows; it also makes your solution available to users on other platforms, such as Outlook on Mac, on mobile, or on the web.

Reviewer: ```suggestion
Because the new Outlook on Windows is based on the Outlook on the web codebase, VSTO and COM add-ins aren't supported. To ensure your add-in continues to work in the new Outlook on Windows, you must migrate your VSTO or COM add-in to an Outlook web add-in. Migrating to an Outlook web add-in not only enables compatibility with the new Outlook on Windows, it also makes your solution available to users on other platforms, such as Outlook on Mac, on mobile, or on the web.
```

- For guidance on how to transition your VSTO add-in to an Outlook web add-in, see [VSTO add-in developer's guide](../overview/learning-path-transition.md) and [Tutorial: Share code between both a VSTO Add-in and an Office Add-in with a shared code library](../tutorials/migrate-vsto-to-office-add-in-shared-code-library-tutorial.md).
- If you're new to Outlook web add-ins, try out the [Outlook quick start](../quickstarts/outlook-quickstart.md) to build your first add-in.

> [!NOTE]
>
> VSTO and COM add-ins are still supported in classic Outlook on Windows.

### Supported scenarios in Outlook web add-ins

The development of the Outlook JavaScript API used by Outlook web add-ins is focused on closing the gap on scenarios that are only supported by VSTO and COM add-in solutions. This way, users who transition to the Outlook web add-in can continue to have a seamless experience.

The following table identifies key Outlook scenarios and their support status in a web add-in. This table will be updated as additional scenarios are supported. Periodically check this section as you plan to migrate your VSTO or COM add-in.

> |Scenario|Description|Support status in Outlook web add-ins|

Reviewer: Consider linking each supported scenario (and perhaps also partially supported ones) to at least one related article or API where possible. That could help readers learn more about what we offer for the specific scenario.

Author: Thanks for the feedback on this. We have this planned for a future iteration while the team works on closing the gaps.

|-----|-----|-----|

> |Spam email reporting and education|Enable users to report unsolicited and potentially unsafe messages and learn how to identify these messages.|Supported. Improvements are in development to further enhance the user experience.|

Reviewer: Consider alphabetizing the scenarios

Author: I'll take this into consideration for a future iteration as the current order is based on most popular scenarios.

|Online meetings|Enable users to create and join online meetings.|Supported.|
|Meeting enhancements|Provide additional services for users when they schedule meetings, such as location selection, catering services, and room lighting and temperature adjustments.|Supported.|
|Online signatures|Automatically add themed signatures to messages and appointments.|Supported.|
|Customer relationship management (CRM) and tracking services|Enable users to send and retrieve information from their CRM system to track communications with existing and potential customers.|Supported.|

> |Content reuse|Enable users to transfer and retrieve text and other content types from third-party systems.|Supported.|

Reviewer: IIRC, instead of "third-party", we should use "partner" or similar term. @lindalu-MSFT, can you advise here?
```suggestion
|Content reuse|Enable users to transfer and retrieve text and other content types from partner systems.|Supported.|
```

|Mail item transformation|Enable users to transform mail items into other formats.|Supported.|

> |Project management|Enable users to create and track project work items from third-party systems.|Supported.|

Author: ```suggestion
|Project management|Enable users to create and track project work items from partner systems.|Supported.|
```

> |Attachment management|Enable users to import or export attachments from third-party locations.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|

Author: ```suggestion
|Attachment management|Enable users to import or export attachments from partner locations.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
```

> |Attachment management|Enable users to import or export attachments from third-party locations.|Partially supported. Essential features are yet to be addressed to create a similar experience to the VSTO or COM add-in.|

Reviewer: ```suggestion
|Attachment management|Enable users to import or export attachments from third-party locations.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
```

|Message encryption|Enable users to encrypt and decrypt messages.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
|Data loss prevention|Prevent users from forwarding mail items that contain highly sensitive information.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
|Mail item classification|Enable users to identify and classify messages that contain sensitive information.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|

> |Data sync service|Enable bidirectional synchronization of mail items with third-party systems.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|

Author: ```suggestion
|Data sync service|Enable bidirectional synchronization of mail items with partner systems.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
```

> |Message encryption|Enable users to encrypt and decrypt messages.|Partially supported. Essential features are yet to be addressed to create a similar experience to the VSTO or COM add-in.|
> |Data loss prevention|Prevent users from forwarding mail items that contain highly sensitive information.|Partially supported. Essential features are yet to be addressed to create a similar experience to the VSTO or COM add-in.|
> |Mail item classification|Enable users to identify and classify messages that contain sensitive information.|Partially supported. Essential features are yet to be addressed to create a similar experience to the VSTO or COM add-in.|
> |Data sync service|Enable bidirectional synchronization of mail items with third-party systems.|Partially supported. Essential features are yet to be addressed to create a similar experience to the VSTO or COM add-in.|

Author: ```suggestion
|Message encryption|Enable users to encrypt and decrypt messages.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
|Data loss prevention|Prevent users from forwarding mail items that contain highly sensitive information.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
|Mail item classification|Enable users to identify and classify messages that contain sensitive information.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
|Data sync service|Enable bidirectional synchronization of mail items with third-party systems.|Partially supported. Essential features are yet to be addressed to create a similar experience to VSTO or COM add-ins.|
```

|Proofing mail items|Provide users with real-time proofreading assistance as they compose messages.|Not currently supported.|

> There are various possibilities to extend the Outlook functionality through add-ins. If your VSTO or COM add-in solution doesn't quite fit any of the scenarios in the table, [complete the survey to share your scenario](https://aka.ms/DevNewOutlook).

Reviewer: ```suggestion
There are various possibilities for extending the Outlook functionality through add-ins. If your VSTO or COM add-in solution doesn't quite fit any of the scenarios in the table, [complete the survey to share your scenario](https://aka.ms/DevNewOutlook).
```

## Support for classic Outlook on Windows

The classic Outlook on Windows desktop client will continue to support the development of new and existing Outlook web add-ins. Additionally, it will continue to receive releases of the latest Outlook add-in features.

## Test your add-in in the new Outlook on Windows

> Test your Outlook web add-in in the new Outlook on Windows today! To switch to the new Outlook on Windows that's in preview, you must:

Reviewer: ```suggestion
Test your Outlook web add-in in the new Outlook on Windows today! To switch to the new Outlook on Windows that's in preview, you must meet the following requirements.
```

> - Have an Exchange-backed Microsoft 365 work or school account. The new client doesn't currently support on-premises, hybrid, or sovereign Exchange accounts.

Reviewer: Does this mean then that, for now, Exchange Online is needed, not just Exchange?
```suggestion
- Have a Microsoft 365 work or school account connected to Exchange Online. The new client doesn't currently support on-premises, hybrid, or sovereign Exchange accounts.
```

- Have a minimum OS installation of Windows 10 Version 1809 (Build 17763).

- Be a member of the [Office Insider Program](https://insider.office.com/join/windows).

> To help you sign up and install the desktop client, see [Getting started with the new Outlook for Windows](https://support.microsoft.com/office/656bb8d9-5a60-49b2-a98b-ba7822bc7627).

Reviewer: ```suggestion
To help you sign up and install the Outlook desktop client, see [Getting started with the new Outlook for Windows](https://support.microsoft.com/office/656bb8d9-5a60-49b2-a98b-ba7822bc7627).
```
